polaris/polaris/utils.py

```diff
@@ -420,18 +412,40 @@ def create_transaction_envelope(transaction, source_account) -> TransactionEnvel
         source_account=source_account,
         network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE,
         base_fee=base_fee,
-    ).append_payment_op(
-        destination=transaction.stellar_account,
-        asset_code=transaction.asset.code,
-        asset_issuer=transaction.asset.issuer,
-        amount=str(payment_amount),
-        source=transaction.asset.distribution_account,
     )
+    if transaction.status == Transaction.STATUS.pending_trust:
+        claimant = Claimant(destination=transaction.stellar_account)
+        asset = Asset(code=transaction.asset.code,
+                      issuer=transaction.asset.issuer)
+        builder.append_create_claimable_balance_op(
+            claimants=[claimant],
+            asset=asset,
+            amount=str(payment_amount),
+            source=transaction.asset.distribution_account,
+        )
+    else:
+        builder.append_payment_op(
+            destination=transaction.stellar_account,
+            asset_code=transaction.asset.code,
+            asset_issuer=transaction.asset.issuer,
+            amount=str(payment_amount),
+            source=transaction.asset.distribution_account,
+        )
     if memo:
         builder.add_memo(memo)
     return builder.build()
 
 
+def get_balanceid(response):
+    claimable_balance_res = response['result_xdr']
+    tr_xdr = TransactionResult.from_xdr(claimable_balance_res)
+    for tr in tr_xdr.result.results:
```

Comment:
Can we return none explictly if we don't find what we're looking for? Also, should this function return none or raise an error? Will developers call this for transactions they know have a claimable balance op or will they use this to determine if one exists?

Reply:
Technically `get_balance_id()` Returns `None` when the loop has exhausted.

Developers are welcome to use this helper method to parse any responses from `submit_transaction(envelope)` to find if they contain `createClaimableBalanceResult`s.

Reply:
I know it _can_ return `None`, but it doesn't _look_ like it does.